mobile/lib/pages/editing/crop.page.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/material.dart';
+import 'package:crop_image/crop_image.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:immich_mobile/utils/hooks/crop_controller_hook.dart'; 
+import 'edit.page.dart';
+
+/// A widget for cropping an image.
+/// 
+/// This widget uses [HookWidget] to manage its lifecycle and state. It allows
+/// users to crop an image and then navigate to the [EditImagePage] with the
+/// cropped image.
+class CropImagePage extends HookWidget {
+  
+  final Image image;
+  const CropImagePage({super.key, required this.image});
+
+  @override
+  Widget build(BuildContext context) {
+    final cropController = useCropController();
+    final aspectRatio = useState<double?>(null);
+
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Colors.black,
+        leading: const CloseButton(color: Colors.white),
+        actions: [
+          IconButton(
+            icon: const Icon(Icons.done_rounded, color: Colors.white, size: 24),
+            onPressed: () async {
+              Image croppedImage = await cropController.croppedImage();
+              Navigator.push(
+                context,
+                MaterialPageRoute(builder: (context) => EditImagePage(image: croppedImage)),
+              );
```

Comment:
We use `auto_route` for routing, so you will need to add `EditImagePage` to the routes by annotating the class with `@RoutePage()`, e.g.:

![image](https://github.com/user-attachments/assets/b2bec3f4-ac8c-47e7-bea9-3c75dab43c92)

Then regenerate the routes:

```bash
fvm flutter pub run build_runner build --delete-conflicting-outputs
```

And then you can route using `context.pushRoute`, e.g.:

![image](https://github.com/user-attachments/assets/bb03fa6f-4a32-475b-aacd-f74bb168224d)

Reply:
Done